### PR TITLE
Give release artifacts table headers

### LINKED ISSUE #2558: Release artifacts table is missing column headers
We had a user who forgot/didn't know to set a `name` attribute for their release artifacts. This led to confusion when debugging source maps, because they weren't sure if they were looking at a `filename` attribute or a `name` attribute on the "Artifacts" page.

It's also possible (but unlikely) to specify a `name` with an empty string, which is confusing in the UI.

Screenshot (first row has no name):

image

I think it wouldn't hurt to be more explicit here and add column headers. This isn't a frequently-visited page (basically only for debugging), so the aesthetic implications (if there are any) seem unimportant.

### PR DESCRIPTION
**Pull request by mattrobenolt**
Also makes files without names appear as "(empty)"

Fixes GH-2558

image

/cc @ckj

### CODE REVIEW

**Comment by dcramer** [comment]
Files should have names. We should address the root cause here.


On January 29, 2016 at 5:44:56 PM, Matt Robenolt (notifications@github.com) wrote:

Also makes files without names appear as "(empty)"

Fixes GH-2558



/cc @ckj

You can view, comment on, or merge this pull request online at:

  https://github.com/getsentry/sentry/pull/2622

Commit Summary

Give release artifacts table headers
File Changes

M src/sentry/static/sentry/app/views/releaseArtifacts.jsx (28)
M src/sentry/static/sentry/less/releases.less (6)
Patch Links:

https://github.com/getsentry/sentry/pull/2622.patch
https://github.com/getsentry/sentry/pull/2622.diff

**Comment by mattrobenolt** [comment]
@dcramer Are you suggesting that we don't need/want the column headers here at all? Or just handling of the "(empty)" case doesn't need to exist? The root cause is that our API explicitly says that "name" is optional.

> The optional 'name' attribute should reflect the absolute path
> that this file will be referenced as. For example, in the case of
> JavaScript you might specify the full web URI.

Not sure what the value is to allow files without names.

**Comment by dcramer** [comment]
The name isn't optional. We dont require the name because in theory the name could be the system filename. In that case the API should error out saying "Name must be specified".

tl;dr names are not optional, and files with empty names are considered invalid (and could be deleted)

**Comment by mattrobenolt** [comment]
@dcramer I added a check that name is not empty on upload. What do we do about existing files?

**Comment by dcramer** [comment]
@mattrobenolt can probably leave it up to individual installs (likely only affects .com). I'm fine w/ us pruning them.

**Comment by mattrobenolt** [comment]
I split out the file name constraint into #2626.

Do we still want to add headers here? Or should I abandon this PR?

**Comment by dcramer** [comment]
no strong opinion either way -- i dont thin they're overly valuable but i dont think its bad either

**Comment by mattrobenolt** [comment]
Alright, gonna go with it because I assume we'll likely want to show more information here at some point I'd imagine.
